targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/TARGET_MCU_K64F/mbed_crc_api.c

```diff
@@ -0,0 +1,101 @@
+#include "crc_api.h"
+
+#include "drivers/fsl_crc.h"
+#include "platform/mbed_assert.h"
+
+#ifdef DEVICE_CRC
+
+static crc_bits_t width;
+
+bool hal_crc_is_supported(const uint32_t polynomial)
+{
+  switch (polynomial)
+  {
+    case POLY_16BIT_CCITT:
+    case POLY_16BIT_IBM:
+    case POLY_32BIT_ANSI:
+      return true;
+    case POLY_8BIT_CCITT:
+    case POLY_7BIT_SD:
+      return false;
+    default:
+      return false;
+  }
+}
+
+void hal_crc_compute_partial_start(const uint32_t polynomial)
+{
+  crc_config_t config;
+
+  switch (polynomial)
+  {
+    case POLY_32BIT_ANSI:
+    {
+      width = kCrcBits32;
+
+      config.polynomial         = polynomial;
+      config.seed               = 0xFFFFFFFFU;
+      config.reflectIn          = true;
+      config.reflectOut         = true;
+      config.complementChecksum = true;
+      config.crcBits            = width;
+      config.crcResult          = kCrcFinalChecksum;
+
+      break;
+    }
+    case POLY_16BIT_IBM:
+    {
+      width = kCrcBits16;
+
+      config.polynomial         = polynomial;
+      config.seed               = 0;
+      config.reflectIn          = true;
+      config.reflectOut         = true;
+      config.complementChecksum = false;
+      config.crcBits            = width;
+      config.crcResult          = kCrcFinalChecksum;
+
+      break;
+    }
+    case POLY_16BIT_CCITT:
+    {
+      width = kCrcBits16;
+
+      config.polynomial         = polynomial;
+      config.seed               = 0xFFFFFFFFU;
+      config.reflectIn          = false;
+      config.reflectOut         = false;
+      config.complementChecksum = false;
+      config.crcBits            = width;
+      config.crcResult          = kCrcFinalChecksum;
+
+      break;
+    }
+    default:
+      MBED_ASSERT("Configuring Mbed CRC with unsupported polynomial");
```

Comment:
Provide an option to set these configs and not ASSERT, it should be user configurable to have 16BIT_CCIIT but seed as Zero. I am sure it will work perfectly fine with HW.

Reply:
As above, it was decided to only support the named subset of polynomials in the enum which define the reflection/seed settings to keep the configuration as simple as possible.

Reply:
You can overload functions to set those parameters. I don;t think sticking to only specified polynomials make sense for HW. In case of software we need to add multiple tables for each polynomial which is not the case in HW.
Also with the same polynomial even SW CRC gives option to have different seed / reflection settings. How are you planning to support this constructor https://github.com/ARMmbed/mbed-os/blob/master/drivers/MbedCRC.h#L127

Also I have strong use case for this requirement, SD driver uses standard CCITT 16-bit but with seed value of 0. https://github.com/ARMmbed/sd-driver/blob/master/SDBlockDevice.cpp#L251

Reply:
> I don;t think sticking to only specified polynomials make sense for HW
I don't either. But then, the only platforms you can reasonably set a polynomial on is K64F and STM(3 series). 

> How are you planning to support this constructor
By disabling the hardware mode when not using the default constructor.

If there's a use case for additional configuration, then it should be considered. @bulislaw

Reply:
The CRC support in HW is rather modest. I don't really like the idea of complicating the API for K64F only for CRC polynomials that are not a very standard ones. Where the HW support actually matters is when we need to calculate the checksum repeatedly and fast eg network packets, bootloader maybe, but that should be using standard polynomials.

Reply:
@bulislaw - Polynomials supported in software CRC are not the only standard ones. Please see the list on Wiki "https://en.wikipedia.org/wiki/Cyclic_redundancy_check" for list.